opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1056,25 +1043,28 @@ namespace Opm {
                         const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
                         const auto& fs = intQuants.fluidState();
                         const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-                        hcpv[region] += pv[c] * hydrocarbon;
-                        pres[region] += pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value();
+                        const double pv = intQuants.porosity().value()*ebosModel.dofTotalVolume(c);
```

Comment:
note that with this approach, rock compressibility is now taken into account here.